solr/core/src/java/org/apache/solr/cloud/overseer/ReplicaMutator.java

```diff
@@ -362,12 +362,7 @@ private ZkWriteCommand updateState(final ClusterState prevState, ZkNodeProps mes
 
     DocCollection newCollection = CollectionMutator.updateSlice(collectionName, collection, slice);
     log.debug("Collection is now: {}", newCollection);
-    if (collection != null && collection.isPerReplicaState()) {
-      PerReplicaStates prs = PerReplicaStates.fetch(collection.getZNode(), zkClient, collection.getPerReplicaStates());
-      return new ZkWriteCommand(collectionName, newCollection, PerReplicaStatesOps.flipState(replica.getName(), replica.getState(), prs), persistCollectionState);
-    } else{
-      return new ZkWriteCommand(collectionName, newCollection);
-    }
+    return new ZkWriteCommand(collectionName, newCollection);
```

Comment:
that looks good!

Reply:
@patsonluk the race condition is here. solrcore sends message to update the replica status to`down`. but it doesn't wait for status update. Then solrcore recovers the core and update the prs state to `up`. 

But sometime overseer node updates the prs state to `down` after that. And then we see core as a down.

With this fix, overseer node will never update the prs state.